server/src/main/java/org/opensearch/tasks/TaskCancellation.java

```diff
@@ -54,7 +55,26 @@ public void cancel() {
         }
 
         task.cancel(getReasonString());
+        runOnCancelCallbacks();
+    }
+
+    /**
+     *  Cancels the task and its descendants and invokes all onCancelCallbacks.
+     */
+    public void cancelTaskAndDescendants(TaskManager taskManager) {
+        if (isEligibleForCancellation() == false) {
+            return;
+        }
+
+        taskManager.cancelTaskAndDescendants(task, getReasonString(), false, ActionListener.wrap(() -> {}));
+        task.cancel(getReasonString());
```

Comment:
Why do we need to call `task.cancel(...)` when we have already called `taskManager.cancalTaskAndDescendants(task, ...)`?

Reply:
For the mocked tasks, we do not register cancellable tasks using `TaskManager#registerCancellableTask` and because of which it returns null [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/tasks/TaskManager.java#L258) and do not call `task.cancel()`.
I tried to call `TaskManager#registerCancellableTask` for the mocked tasks as well but could not mock `cancellableTasks`. Let me know if you have any other workaround.

Reply:
The fix should be in unit tests instead – either registering the cancellable task, or by mocking the task manager's cancel method.

Reply:
Done.